arch/sim/src/sim/up_oneshot.c

```diff
@@ -357,6 +358,16 @@ FAR struct oneshot_lowerhalf_s *oneshot_initialize(int chan,
 
 void up_timer_initialize(void)
 {
+  /* Block the signals for the new threads created on the host to prevent
+   * a race condition where the simulated interrupt handler runs on another
+   * host thread. The new threads will inherit the signal mask which has
+   * blocked signals.
+   */
+
+#ifdef CONFIG_SIM_PREEMPTIBLE
+  up_prepare_timer();
```

Comment:
As I said before, timer isn't compatible with oneshot timer, both can't work together. you should integrate the alarm timer logic into oneshot.

Reply:
Thanks for the suggestion, ```up_oneshot.c``` is part of the CSRCS list and it's using the NuttX API. My timer is using the host timer and it builds in the HOSTSRC list. I am not sure if merging them it's ok. What do you think ?

Reply:
Yes, up_oneshot.c is the bridge between NuttX and host. up_oneshot.c implmeent the standard NuttX timer driver interface and forward the real timing call(host_sleepuntil) to up_hosttime.c which is part of HOSTSRC. Here is the basic idea to integrate your code into up_oneshot.c:
1.Implment the alaram wrapper and signal handler in up_hosttime.c
2.These functions should forward to your alarm wrapper in up_hosttime.c:
  oneshot_initialize
  sim_start
  sim_cancel
  it's almost 1:1 mapping for alarm API
3.Signal handler should call to some function(similar with up_timer_update) in up_hosttime.c
Of course, we should make sure the code can run without problem with/without CONFIG_SIM_PREEMPTIBLE.